### PR TITLE
[dynamo] cast single-elem tensors to float and int 

### LINKED ISSUE #109538: [dynamo] torch._dynamo.exc.Unsupported: call_function BuiltinVariable(float) [TensorVariable()] {}
### 🐛 Describe the bug

self explanatory

from exporting emformer at https://github.com/pytorch/pytorch/issues/107188

```
from user code:
   File "/data/users/ezyang/a/torchaudio/torchaudio/models/rnnt_decoder.py", line 293, in forward
    return self._search(enc_out, None, beam_width)
  File "/data/users/ezyang/a/torchaudio/torchaudio/models/rnnt_decoder.py", line 246, in _search
    b_hypos = self._gen_b_hypos(b_hypos, a_hypos, next_token_probs, key_to_b_hypo)
  File "/data/users/ezyang/a/torchaudio/torchaudio/models/rnnt_decoder.py", line 155, in _gen_b_hypos
    score = float(append_blank_score)
```

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### PR DESCRIPTION
**Pull request by jon-chuang**
Fixes https://github.com/pytorch/pytorch/issues/109538

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -589,6 +589,26 @@ def call_function(
             )
             return out
 
+        # TODO(jon-chuang): Handle complex, bool
+        if self.fn in (int, float) and isinstance(args[0], variables.TensorVariable):
```
Comment (jon-chuang) [inline torch/_dynamo/variables/builtin.py:593]: Could you combine the logic with above?

File: test/dynamo/test_functions.py [hunk 2]
```diff
@@ -1931,6 +1931,34 @@ def fn(x):
...
+                opt_fn = torch.compile(
+                    fn, backend="eager", fullgraph=True, dynamic=False
+                )
+                x = torch.tensor([val])
+                res = fn(x)
+                ref = opt_fn(x)
+                self.assertEqual(ref, res)
+
+                with self.assertRaises(ValueError):
+                    fn(torch.tensor([val] * 2))
```
Comment (jon-chuang) [inline test/dynamo/test_functions.py:1958]: Add comment cannot handle non single elem

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**